tests/Aspire.Hosting.Tests/Publishing/ResourceContainerImageBuilderTests.cs

```diff
@@ -643,6 +643,153 @@ public async Task BuildImageAsync_ThrowsInvalidOperationException_WhenDockerRunt
         Assert.Contains(logs, log => log.Message.Contains("Container runtime is not running or is unhealthy. Cannot build container images."));
     }
 
+    [Fact]
+    [RequiresDocker]
+    public async Task CanBuildImageFromDockerfileWithBuildArgsSecretsAndStage()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create(output);
+
+        builder.Services.AddLogging(logging =>
+        {
+            logging.AddFakeLogging();
+            logging.AddXunit(output);
+        });
+
+        // Create a fake container runtime to capture build arguments and secrets
+        var fakeContainerRuntime = new FakeContainerRuntime(shouldFail: false);
+        builder.Services.AddKeyedSingleton<IContainerRuntime>("docker", fakeContainerRuntime);
+
+        var (tempContextPath, tempDockerfilePath) = await DockerfileUtils.CreateTemporaryDockerfileAsync();
+
+        // Add parameters for build args and secrets
+        builder.Configuration["Parameters:goversion"] = "1.22";
+        builder.Configuration["Parameters:secret"] = "mysecret";
+
+        var goVersionParam = builder.AddParameter("goversion");
+        var secretParam = builder.AddParameter("secret", secret: true);
+
+        var container = builder.AddDockerfile("container", tempContextPath, tempDockerfilePath)
+                              .WithBuildArg("GO_VERSION", goVersionParam)
+                              .WithBuildArg("STATIC_ARG", "static-value")
+                              .WithBuildSecret("SECRET_ASENV", secretParam);
```

Comment:
It would be valuable to have a test where the secret has a `null` value.

Reply:
This ends up being an interesting one since `WithBuildSecret` only exposes a single overload that takes a `ParameterResource`. I updated the implementation to fallback to reading unresolved parameter values from the environment variable by not emitting a value. In a typical deployment case, we'd prompt for these parameters upfront.